### PR TITLE
modify DataServlet to return hard-coded JSON data and fetch that JSON

### PR DESCRIPTION
**Pull request by tbarnahor**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -24,9 +27,33 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+    private List<String> messages;
+
+  @Override
+  public void init() {
+    messages = new ArrayList<>();
```
Comment (arielz-g) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: You can initialize the list with the values already:
new ArrayList<>(Arrays.asList("a", "b", "c"));
Or alternatively use messages.addAll(Arrays.asList(..));

Generally speaking if you notice you're repeating the same method several times in a bulk, there's a use case for an aggregation of some sort and there's probably a way to do so:)

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -24,9 +27,33 @@
...
+    messages.add("Have a great day :)");
+    messages.add("Hi there!");
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello Tal!");
+    // Convert messages to JSON
+    String json = convertToJson(messages);
```
Comment (arielz-g) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: No need for a parameter here, you can do this in place.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -24,9 +27,33 @@
...
+
+    // Send the JSON as the response
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
   }
+
+
+  private String convertToJson(List messages) {
```
Comment (arielz-g) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: This method is redundant. Notice you can just do response.getWriter().println(new Gson().toJson(messages)), and have yourself a one-liner (if you prefer you can split the "new Gson()" to a param, but overall I'd keep it in doGet.

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -50,10 +50,21 @@ function createImg(imageName) {
 
 
 function getHelloMsg(){
-    fetch('/data').then(response => response.text()).then((massage) => {
-    document.getElementById('massage-container').innerText = massage;
+    fetch('/data').then(response => response.json()).then((massages) => {
+    const messageListElement = document.getElementById('massage-container');
+    messageListElement.innerText = '';
+    messageListElement.appendChild(createListElement(massages[2]));
```
Comment (arielz-g) [inline portfolio/src/main/webapp/script.js]: Now noticed that you actually want to use the messages you're inputting in initialization here.
This however isn't good practice: the reader doesn't know what indexes 2, 0 and 1 stand for, and imagine these change? Creating this type of dependency is risky. Also I believe you'd like to update your messages at some point, so this has to be generalized.

I'd suggest iterating on the messages (with a loop) and adding the messages one after the other. This means you'd have to keep them in order in your constructor as well, yet this is far less error prone.

Comment (tbarnahor) [reply portfolio/src/main/webapp/script.js]: Done

Comment (arielz-g) [reply portfolio/src/main/webapp/script.js]: The changes you made aren't pushed (still looking at the previous version).

**Comment by tbarnahor** [comment]
> Relying on a number is also bug-prone here. Why not use messages.length? or better yet something like:
> messages.forEach(message) => {messageListElement.appendChild(..message)}

Got it, thanks

**Review by arielz-g — APPROVED** [review]

**Pull request merged**